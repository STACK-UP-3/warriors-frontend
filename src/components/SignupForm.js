import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { onSignup } from '../redux/actions/signupAction';
import SyncLoader from 'react-spinners/SyncLoader';
import { toast, ToastContainer } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';

import M from 'materialize-css'; // Materialize JS ~ https://www.youtube.com/watch?v=EboLM8OjlP4

export class SignupForm extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  handleChange = (event) => {
    this.setState({
      [event.target.name]: event.target.value,
    });
  };

  handleSignup = async (event) => {
    event.preventDefault();
    const signupData = {
      firstname: this.state.firstname,
      lastname: this.state.lastname,
      email: this.state.email,
      password: this.state.password,
    };
    await this.props.signupProcess(signupData);
    toast.info(this.props.signup.message);
    this.props.signup.error
      ? null
      : document.getElementById('signupForm').reset();
  };

  onCloseToast = () => {
    window.location.replace('/');
  };

  render() {
    return (
      <div>
        <form
          id='signupForm'
          className='col s12'
          testdata='signupForm'
          onSubmit={this.handleSignup}>
          <div className='form-content'>
            <div className='row' id='textInput'>
              <div className='input-field col s12'>
                <input
                  name='firstname'
                  id='firstname'
                  type='text'
                  className='validate'
                  testdata='inputChange'
                  onChange={this.handleChange}
                />
                <label htmlFor='firstname'>First Name</label>
              </div>
            </div>
            <div className='row' id='textInput'>
              <div className='input-field col s12'>
                <input
                  name='lastname'
                  id='lastname'
                  type='text'
                  className='validate'
                  testdata='inputChange'
                  onChange={this.handleChange}
                  disabled={this.props.signup.loading}
                />
                <label htmlFor='lastname'>Last Name</label>
              </div>
            </div>

            <div className='row' id='textInput'>
              <div className='input-field col s12'>
                <input
                  name='email'
                  id='email'
                  type='email'
                  className='validate'
                  testdata='inputChange'
                  onChange={this.handleChange}
                  disabled={this.props.signup.loading}
                />
                <label htmlFor='email'>Email</label>
              </div>
            </div>

            <div className='row' id='textInput'>
              <div className='input-field col s12'>
                <input
                  name='password'
                  id='password'
                  type='password'
                  className='validate'
                  onChange={this.handleChange}
                  disabled={this.props.signup.loading}
                />
                <label htmlFor='password'>Password</label>
              </div>
            </div>

            <div className='contentFooter'>
              {this.props.signup.error ? (
                <p className='error'>{this.props.signup.message}</p>
              ) : (
                <ToastContainer
                  position='top-center'
                  autoClose={false}
                  hideProgressBar={true}
                  newestOnTop={false}
                  closeOnClick
                  rtl={false}
                  pauseOnFocusLoss
                  draggable={false}
                  pauseOnHover
                  onClick={this.onCloseToast}
                />
              )}
              <p>
                Have an account already? <a href='/login'>Login</a>
              </p>
              <br></br>
            </div>
          </div>

          <div className='form-footer'>
            <div className='submitBtn'>
              <button
                className='btn waves-effect waves-light'
                type='submit'
                id='submit'
                disabled={
                  !this.state.firstname ||
                  !this.state.lastname ||
                  !/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(
                    this.state.email,
                  ) ||
                  !this.state.password
                }>
                {this.props.signup.loading ? (
                  <SyncLoader
                    size={5}
                    margin={5}
                    color={'#fff'}
                    loading={this.props.signup.loading}
                  />
                ) : (
                  'Signup'
                )}
              </button>
            </div>
          </div>
        </form>
      </div>
    );
  }
}

SignupForm.propTypes = {
  signupProcess: PropTypes.func.isRequired,
};

const mapStateToProps = (response) => ({ signup: response.signup });
const mapDispatchToProps = (dispatch) => ({
  signupProcess: (signupData) => dispatch(onSignup(signupData)),
});

export default connect(mapStateToProps, mapDispatchToProps)(SignupForm);
