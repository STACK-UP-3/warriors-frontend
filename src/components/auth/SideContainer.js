import React from 'react';

export default (props) => (
  <div className='sideContainer'>
    <div className='background'></div>

    <div className='content'>
      <h3>{props.title}</h3>
      {props.subtitle}
    </div>
  </div>
);
