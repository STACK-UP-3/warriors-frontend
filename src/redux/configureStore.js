import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import promise from 'redux-promise-middleware';
import { composeWithDevTools } from 'redux-devtools-extension';
import reducersCombination from './reducers/index';



export default ()=>{
    const store = createStore(
        reducersCombination, {}, 
        composeWithDevTools(applyMiddleware(thunk,promise))
    );
    
    return store;
}
