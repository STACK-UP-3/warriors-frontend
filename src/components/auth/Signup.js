import React, { Component } from 'react';
import SignupForm from './SignupForm';
import SideContainer from './SideContainer';
import Footer from '../common/Footer';

const signupTitle = 'Welcome to BareFoot Nomad';
const signupSubTitle = (
  <div>
    <p>
      Join 3000+<br></br> people using<br></br> Barefoot Nomad
    </p>
    <br />
  </div>
);

export class Signup extends Component {
  render() {
    return (
      <div className='signupContainer'>
        <div className='main'>
          <SideContainer title={signupTitle} subtitle={signupSubTitle} />
          <div className='formContainer'>
            <h4 className='center'> Sign up to BareFoot Nomad </h4>
            <br></br>
            <SignupForm testdata='signupForm' />
            <br></br>
            <p className='center-align'>Or Signup with:</p>
            <br></br>
            <div className='socialAuth container center'>
              <a href=''>
                <img id='google' src='../assets/google.svg' />
              </a>
              <a href=''>
                <img id='facebook' src='../assets/facebook.png' />
              </a>
            </div>
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}

export default Signup;
