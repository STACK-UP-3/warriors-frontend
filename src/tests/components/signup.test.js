import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import mockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Signup from '../../components/auth/Signup';
import SignupFormD from '../../components/auth/SignupForm';
import configureStore from '../../redux/configureStore';
import { SignupForm } from '../../components/auth/SignupForm';

describe('Test <SignupForm/> Component', () => {
  const store = configureStore();
  let onFormSubmit;
  beforeEach(() => {
    onFormSubmit = jest.fn();
  });

  it('Should render the signup form correctly', () => {
    const store = mockStore([thunk])({
      signup: {
        loading: false,
        error: false,
        message: '',
      },
    });

    const wrapper = mount(
      <Provider store={store}>
        <SignupFormD />
      </Provider>,
    );

    const handleSubmit = jest.spyOn(
      wrapper.find('[testdata="signupForm"]').props(),
      'onSubmit',
    );
    wrapper.find('[testdata="signupForm"]').props().onSubmit();
    expect(wrapper).toMatchSnapshot();
    expect(handleSubmit).toHaveBeenLastCalledWith();
  });

  it('Should test form inputs on change', () => {
    const testState = { email: 'aaa@bbbb.ccc', password: 'jjjjjjj' };
    const wrapper = mount(
      <Provider store={store}>
        <SignupFormD />
      </Provider>,
    );

    wrapper
      .find('input')
      .at(0)
      .simulate('change', {
        target: { id: 'firstname', value: 'name', name: 'firstname' },
      });

    wrapper
      .find('input')
      .at(1)
      .simulate('change', {
        target: { id: 'lastname', value: 'lastnamr', name: 'lastname' },
      });
    wrapper
      .find('input')
      .at(2)
      .simulate('change', {
        target: { id: 'email', value: 'xxx@yyy.zzz', name: 'email' },
      });

    wrapper
      .find('input')
      .at(3)
      .simulate('change', {
        target: { id: 'password', value: 'kkkkkkkk', name: 'password' },
      });

    expect(wrapper.length).toBe(1);
  });

  it('Should render signup page', () => {
    const page = shallow(<Signup />);
    expect(page).toMatchSnapshot();
  });

  it('Should handle signup process', () => {
    const signup = shallow(
      <SignupForm signupProcess={jest.fn()} signup={{ loading: false }} />,
    );
    const env = {
      preventDefault: jest.fn(),
    };
    const instance = signup.instance();
    instance.setState({
      firstname: 'first',
      lastname: 'lassr',
      email: 'email@yahoo.com',
      password: 'password123',
    });
    instance.handleSignup(env);
    instance.onCloseToast();
  });
});
