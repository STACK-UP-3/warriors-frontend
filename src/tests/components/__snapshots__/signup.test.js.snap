// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <SignupForm/> Component Should render signup page 1`] = `
<div
  className="signupContainer"
>
  <div
    className="main"
  >
    <_default
      subtitle={
        <div>
          <p>
            Join 3000+
            <br />
             people using
            <br />
             Barefoot Nomad
          </p>
          <br />
        </div>
      }
      title="Welcome to BareFoot Nomad"
    />
    <div
      className="formContainer"
    >
      <h1>
         Sign up to BareFoot Nomad 
      </h1>
      <Connect(SignupForm)
        testdata="signupForm"
      />
      <br />
      <div
        className="socialAuth"
      >
        <p>
          Or Signup with:
        </p>
        <div
          className="socialAuthImg"
        >
          <div
            id="google"
          >
            <img
              src="../assets/google.svg"
            />
          </div>
          <div
            id="facebook"
          >
            <img
              src="../assets/facebook.png"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <_default />
</div>
`;

exports[`Test <SignupForm/> Component Should render the signup form correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(SignupForm)>
    <SignupForm
      signup={
        Object {
          "error": false,
          "loading": false,
          "message": "",
        }
      }
      signupProcess={[Function]}
    >
      <div>
        <form
          className="col s12"
          id="signupForm"
          onSubmit={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": Promise {},
                },
              ],
            }
          }
          testdata="signupForm"
        >
          <div
            className="form-content"
          >
            <div
              className="row"
              id="textInput"
            >
              <div
                className="input-field col s12"
              >
                <input
                  className="validate"
                  id="firstname"
                  name="firstname"
                  onChange={[Function]}
                  testdata="inputChange"
                  type="text"
                />
                <label
                  htmlFor="firstname"
                >
                  First Name
                </label>
              </div>
            </div>
            <div
              className="row"
              id="textInput"
            >
              <div
                className="input-field col s12"
              >
                <input
                  className="validate"
                  disabled={false}
                  id="lastname"
                  name="lastname"
                  onChange={[Function]}
                  testdata="inputChange"
                  type="text"
                />
                <label
                  htmlFor="lastname"
                >
                  Last Name
                </label>
              </div>
            </div>
            <div
              className="row"
              id="textInput"
            >
              <div
                className="input-field col s12"
              >
                <input
                  className="validate"
                  disabled={false}
                  id="email"
                  name="email"
                  onChange={[Function]}
                  testdata="inputChange"
                  type="email"
                />
                <label
                  htmlFor="email"
                >
                  Email
                </label>
              </div>
            </div>
            <div
              className="row"
              id="textInput"
            >
              <div
                className="input-field col s12"
              >
                <input
                  className="validate"
                  disabled={false}
                  id="password"
                  name="password"
                  onChange={[Function]}
                  type="password"
                />
                <label
                  htmlFor="password"
                >
                  Password
                </label>
              </div>
            </div>
            <div
              className="contentFooter"
            >
              <ToastContainer
                autoClose={false}
                closeButton={[Function]}
                closeOnClick={true}
                draggable={false}
                draggablePercent={80}
                hideProgressBar={true}
                newestOnTop={false}
                onClick={[Function]}
                pauseOnFocusLoss={true}
                pauseOnHover={true}
                position="top-center"
                role="alert"
                rtl={false}
                transition={[Function]}
              >
                <div
                  className="Toastify"
                />
              </ToastContainer>
              <p>
                Have an account already? 
                <a
                  href="/login"
                >
                  Login
                </a>
              </p>
              <br />
            </div>
          </div>
          <div
            className="form-footer"
          >
            <div
              className="submitBtn"
            >
              <button
                className="btn waves-effect waves-light"
                disabled={true}
                id="submit"
                type="submit"
              >
                Signup
              </button>
            </div>
          </div>
        </form>
      </div>
    </SignupForm>
  </Connect(SignupForm)>
</Provider>
`;
